src/main/resources/Diagram/DiagramViewSheet.xml

```diff
@@ -319,6 +325,29 @@ require.config({
     diagramViewerDeferred.reject();
   });
 
+  // Return a diagram in XML format if is any present in URL
+  function getDiagramXMLFromURL() {
+    if (window.location.hash !== '') {
+      if (window.location.hash.substring(0, 2) == '#R') {
+        // We need to exlude first 2 letters from hash (#R).
+        let decodedData = decodeURIComponent(window.location.hash.substring(2));
+        var graph = new Graph();
+        let rawDiagram = graph.decompress(decodedData);
+        return mxUtils.parseXml(rawDiagram);
+      }
+    }
+    return null;
+  };
+
+  function getDiagramXML(container) {
+    var xmlDiagramFromURL = getDiagramXMLFromURL();
+    if (null === xmlDiagramFromURL) {
+      rawDiagram = container.data('model');
+      return mxUtils.parseXml(rawDiagram);
```

Comment:
I would leave the parsing outside of ``getDiagramXMLFromURL`` and ``getDiagramXML``. In other words, I would simply return the raw diagram here and do the parsing above when creating the ``GraphViewer``.

Reply:
The reason is because we already have a method to parse the XML (that is ``mxUtils.parseXml()``) so I find it cleaner if ``getDiagramXMLFromURL`` and ``getDiagramXML`` do a single thing: get the raw XML. Then you can do whatever you want with the raw XML (parse it or not).